modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -764,6 +768,28 @@ public List<ObjectDefinition> getSystemObjectDefinitions() {
 		return objectDefinitionPersistence.findBySystem(true);
 	}
 
+	@Transactional(propagation = Propagation.REQUIRES_NEW)
+	public long getUniqueComposedKeyObjectValidationRuleDslQueryCount(
+		DynamicObjectDefinitionTable dynamicObjectDefinitionTable,
+		DynamicObjectDefinitionTable extensionDynamicObjectDefinitionTable,
+		Predicate searchPredicate) {
+
+		return dslQueryCount(
```

Comment:
Question: Are we considering objects scoped by site? Because this query will consider the entries as a mix of the sites.

Reply:
Good point!
I'll check it.